paywall/src/data-iframe/postOffice.js

```diff
@@ -71,8 +72,24 @@ export default function postOffice(window, requiredConfirmations) {
   }
 
   const lockHandler = cachedData => {
-    actions.locks(cachedData.locks)
-    const unlockedLockAddresses = Object.values(cachedData.locks)
+    const relevantLocks = getRelevantLocks()
+    if (!relevantLocks) {
+      // this should not happen
+      console.error('internal error - locks requested before config') // eslint-disable-line
+      return
+    }
+    // we can only return the locks that are specifically requested
+    const filteredLocks = Object.keys(cachedData.locks).reduce(
```

Comment:
How would you feel about extracting this into its own method so that it can be tested robustly outside of `postOffice`?

Reply:
done